JuliaInterface/gap/JuliaInterface.gi

```diff
@@ -57,11 +57,74 @@ InstallMethod( CallFuncList,
     end );
 
 
-##
-##  We support '<juliaobj>[<i>]' and '<juliaobj>[<i>, <j>]' in general.
-##  This is useful for example if <juliaobj> is a tuple.
-##
-BindJuliaFunc( "getindex", "Base" );
+BindGlobal( "_JULIA_MODULE_TYPE", JuliaEvalString( "Module" ) );
+BindGlobal( "_JULIA_FUNCTION_TYPE", JuliaEvalString( "Function" ) );
+BindGlobal( "_JULIA_ISA", JuliaFunction( "isa" ) );
+
+BindGlobal( "_NewJuliaModule",
+  function( name, julia_pointer )
+    local module;
+
+    module := rec( storage := rec( ),
+                   julia_pointer := julia_pointer );
+
+    Objectify( TheTypeOfJuliaModules, module );
+
+    SetName( module, name );
+
+    return module;
+
+end );
+
+InstallMethod( \.,
+              [ IsJuliaModule, IsPosInt ],
+  function( module, rnum )
+    local rnam, global_variable;
+
+    if IsBound\.( module!.storage, rnum ) then
+        return \.(module!.storage, rnum );
+    fi;
+
+    rnam := NameRNam( rnum );
+
+    global_variable := _JuliaGetGlobalVariableByModule( rnam, module!.julia_pointer );
+
+    if ConvertedFromJulia( _JULIA_ISA( global_variable, _JULIA_FUNCTION_TYPE ) ) then
+        global_variable := _JuliaFunction( global_variable );
+    elif ConvertedFromJulia( _JULIA_ISA( global_variable, _JULIA_MODULE_TYPE ) ) then
+        global_variable := _NewJuliaModule( rnam, global_variable );
+    fi;
+
+    \.\:\=( module!.storage, rnum, global_variable );
+    return global_variable;
+
+end );
+
+InstallMethod( \.\:\=,
+               [ IsJuliaModule, IsPosInt, IsObject ],
+  function( module, rnum, obj )
+    \.\:\=( module!.storage, rnum, obj );
+end );
+
+
+InstallMethod( IsBound\.,
+               [ IsJuliaModule, IsPosInt ],
+    {module,rnum} -> IsBound\.( module!.storage, rnum  ) );
+
+
+InstallMethod( Unbind\.,
+               [ IsJuliaModule, IsPosInt ],
+  function( module, rnum )
+    Unbind\.( module!.storage, rnum );
+end );
```

Comment:
Just for curiosity:
Why would one want to unbind symbols?
(One situation would be in the `Main` module if one could unbind Julia variables;
but is this possible?)

Reply:
Well, technically, any import could override variables in `Main`, so `Unbind` could potentially be useful.

Granted, I just added it for completeness ;)

Reply:
I would call `Error` here, just as for assignment.